report-api/src/test/java/no/sikt/nva/data/report/api/fetch/FetchDataReportTest.java

```diff
@@ -45,29 +62,39 @@ void shouldThrowBadRequest(TestingRequest report)
 
     @ParameterizedTest
     @ArgumentsSource(ValidRequestSource.class)
-    void shouldReturnFormattedResult(TestingRequest request) throws IOException {
-        var service = new QueryService(setupDatabaseConnection());
+    void shouldReturnFormattedResult(TestingRequest request) throws IOException, BadRequestException {
+        var testData = new TestData(List.of(new DatePair(new PublicationDate("2023", "02", "02"),
+                                                     Instant.now().minus(100, ChronoUnit.DAYS)),
+                                        new DatePair(new PublicationDate("2023", "10", "18"),
+                                                     Instant.now().minus(100, ChronoUnit.DAYS))));
```

Comment:
This is the crux of the issue, we can now generate full records with differences that allow us to filter them.